File: postgres_39/all_struct_with_default/SeisAcqtnSurvey.go
package dto

type Seis_acqtn_survey struct {
	Seis_set_subtype     string  `json:"seis_set_subtype" default:"source"`
	Seis_acqtn_survey_id string  `json:"seis_acqtn_survey_id" default:"source"`
	Acqtn_survey_name    *string `json:"acqtn_survey_name" default:""`
	Active_ind           *string `json:"active_ind" default:""`
	Area_id              *string `json:"area_id" default:""`
	Area_type            *string `json:"area_type" default:""`
	Completed_date       *string `json:"completed_date" default:""`
	Completed_date_desc  *string `json:"completed_date_desc" default:""`
	Effective_date       *string `json:"effective_date" default:""`
	Expiry_date          *string `json:"expiry_date" default:""`
	Ppdm_guid            *string `json:"ppdm_guid" default:""`
	Remark               *string `json:"remark" default:""`
	Seis_dimension       *string `json:"seis_dimension" default:""`
	Shot_for             *string `json:"shot_for" default:""`
	Source               *string `json:"source" default:""`
	Start_date           *string `json:"start_date" default:""`
	Start_date_desc      *string `json:"start_date_desc" default:""`
	Row_changed_by       *string `json:"row_changed_by" default:""`
	Row_changed_date     *string `json:"row_changed_date" default:""`
	Row_created_by       *string `json:"row_created_by" default:""`
	Row_created_date     *string `json:"row_created_date" default:""`
	Row_effective_date   *string `json:"row_effective_date" default:""`
	Row_expiry_date      *string `json:"row_expiry_date" default:""`
	Row_quality          *string `json:"row_quality" default:""`
}

File: create_tables/seis_acqtn_survey.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table seis_acqtn_survey';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table seis_acqtn_survey(
        SEIS_SET_SUBTYPE VARCHAR(30) NOT NULL,
        SEIS_ACQTN_SURVEY_ID VARCHAR(40) NOT NULL,
        ACQTN_SURVEY_NAME VARCHAR(255),
        ACTIVE_IND VARCHAR(1),
        AREA_ID VARCHAR(40),
        AREA_TYPE VARCHAR(40),
        COMPLETED_DATE TIMESTAMP(0),
        COMPLETED_DATE_DESC VARCHAR(8),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        SEIS_DIMENSION VARCHAR(40),
        SHOT_FOR VARCHAR(40),
        SOURCE VARCHAR(40),
        START_DATE TIMESTAMP(0),
        START_DATE_DESC VARCHAR(8),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")